modules/repository/branch.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repository
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/db"
+	git_model "code.gitea.io/gitea/models/git"
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/builder"
+)
+
+func SyncAllBranches(ctx context.Context, doerID int64) error {
+	log.Trace("Synchronizing repository branches (this may take a while)")
+	return db.Iterate(ctx, builder.Eq{"is_empty": false}, func(ctx context.Context, repo *repo_model.Repository) error {
+		return SyncRepoBranches(ctx, repo.ID, doerID)
```

Comment:
If a single "SyncRepoBranches" fails, the "iterate" should continue to finish other's sync.

I guess if the "err" is returned, the "iterate" would stop and the remaining repos won't get synced.

Reply:
It will. The queue will take the failed sync task back to the queue.

Reply:
No queue here. What do you mean? The queue code is only for initBranchSyncQueue -> addRepoToBranchSyncQueue


But here, if any `SyncRepoBranches` fails, the `Iterate` stops, I do not think it is right.

Reply:
This function is unused, so I removed it.